tensorflow/lite/micro/kernels/lstm_eval_test.cc

```diff
@@ -17,594 +17,1312 @@ limitations under the License.
+            CreateQuantizedGateParameters<WeightType, BiasType>(
+                cell_gate_params,
+                quantization_settings.input_quantization_parameters,
+                quantization_settings.output_quantization_parameters,
+                quantization_settings.cell_gate_quantization_parameters),
+            CreateQuantizedGateParameters<WeightType, BiasType>(
+                output_gate_params,
+                quantization_settings.input_quantization_parameters,
+                quantization_settings.output_quantization_parameters,
+                quantization_settings.output_gate_quantization_parameters)),
+        quantization_settings_(quantization_settings) {
+    // Setup the IntegerLstmParameter
+    AssembleEvalualtionParams();
+    // Initialize hidden state (zero_point)
+  }
 
-  int layer_norm_output_size_[2];
-  TfLiteEvalTensor layer_norm_output_tensor_;
+  const IntegerLstmParameter* GetEvaluationParameters() const {
+    return &evaluation_params_;
+  }
 
-  int input_gate_bias_size_[2];
-  TfLiteEvalTensor input_gate_bias_tensor_;
+  const ModelQuantizationParameters* QuantizationSettings() const {
+    return &quantization_settings_;
+  }
 
-  int forget_gate_bias_size_[2];
-  TfLiteEvalTensor forget_gate_bias_tensor_;
+  // Set the input tensor, quantized version
+  void QuantizeInputTensor(const float* data) {
+    ActivationType quantized_input_data[kInputSize];
+    Quantize(data, quantized_input_data, kInputSize,
+             quantization_settings_.input_quantization_parameters.scale,
+             quantization_settings_.input_quantization_parameters.zero_point);
+    this->SetInputTensorData(quantized_input_data);
+  }
 
-  int cell_gate_bias_size_[2];
-  TfLiteEvalTensor cell_gate_bias_tensor_;
+  // Set the hidden state tensor, quantized version
+  void QuantizeHiddenStateTensor(const float* data) {
+    ActivationType quantized_hidden_state_data[kGateOutputSize];
+    Quantize(data, quantized_hidden_state_data, kGateOutputSize,
+             quantization_settings_.hidden_quantization_parameters.scale,
+             quantization_settings_.hidden_quantization_parameters.zero_point);
+    this->SetHiddenStateTensorData(quantized_hidden_state_data);
+  }
 
-  int output_gate_bias_size_[2];
-  TfLiteEvalTensor output_gate_bias_tensor_;
+ private:
+  // Quantization settings for every tensor inside the model
+  const ModelQuantizationParameters quantization_settings_;
+  // All the information that required to invoke the quantized kernel
+  IntegerLstmParameter evaluation_params_;
 
-  // projection_bias.
-  int32_t projection_bias_[n_output_];
+  void AssembleEvalualtionParams() {
+    double effective_scale;
+    // QuantizeMultiplier takes int as the output shift type, but
+    // the shift type is stored as int32_t inside the IntegerLstmParameter.
+    // Hexagon compilation requires the exact match of the two. Use a buffer to
+    // pass the test (ugly!!!)
```

Comment:
Let's make bug and put a TODO(b/): int vs int32_t on Hexagon has been problematic in the past as well. Perhaps the way forward is to use an int in the IntegerLstmParameter when we start thinking about refactoring the implementation itself?

Reply:
Done